lektor_markdown_header_anchors.py

```diff
@@ -14,32 +22,22 @@ class MarkdownHeaderAnchorsPlugin(Plugin):
     def on_markdown_config(self, config, **extra):
         class HeaderAnchorMixin(object):
             def header(renderer, text, level, raw):
-                anchor = slugify(raw)
-                renderer.meta['toc'].append((level, anchor, Markup(text)))
+                anchor = uniquify(renderer.meta['uniquify'], raw)
+                renderer.meta['toc'].append(TocEntry(level, anchor, Markup(text), []))
                 return '<h%d id="%s">%s</h%d>' % (level, anchor, text, level)
         config.renderer_mixins.append(HeaderAnchorMixin)
 
     def on_markdown_meta_init(self, meta, **extra):
         meta['toc'] = []
+        meta['uniquify'] = Counter()
 
     def on_markdown_meta_postprocess(self, meta, **extra):
-        prev_level = None
-        toc = []
-        stack = [toc]
-
-        for level, anchor, title in meta['toc']:
-            if prev_level is None:
-                prev_level = level
-            elif prev_level == level - 1:
-                stack.append(stack[-1][-1][2])
-                prev_level = level
-            elif prev_level > level:
-                while prev_level > level:
-                    # Just a simple workaround for when people do weird
-                    # shit with headlines.
-                    if len(stack) > 1:
-                        stack.pop()
-                    prev_level -= 1
-            stack[-1].append(TocEntry(anchor, title, []))
-
-        meta['toc'] = toc
+        stack = [TocEntry(-1, None, None, [])]
+
+        for entry in meta['toc']:
```

Comment:
I think it is worth adding a comment here describing what the new behavior does

Reply:
create a dummy TocEntry that will never be removed (header level of -1) this way there is no need to check if the stack is empty. At the end I just return the children of that dummy entry.

Reply:
what's the procedure for updating a pull request? should I close and start a new one?

Reply:
You just keep making commits to it and pushing to your fork...

`change code`
`git add . `
`git commit -m "Nice message"`
`git push origin master` (assuming origin is pointing to your fork.

Reply:
Well i mean that you changed somehow how the logic worked... so we need to add this for whomever comes next and reads :-p


>Bugfix: If the header level jumps by more than one at a time the new header is put at the same level as the previous header instead of below it in the tree. For example, if H4 follows H2, it would be put at the same level as the H2 instead of below it. I also simplified the code a bit.